### ISSUE TITLE
Passing a tuple of paths to upload over SFTP fails

### ISSUE THREAD

**Issue by Lyrositor** [ronf/asyncssh#265]
When uploading multiple files at once using the SFTP client, the call can fail if the paths are passed as a tuple to `SFTPClient.put`. This is because of the code in this section, in `asyncssh/logging.py`:

```python3
    def log(self, level, msg, *args, **kwargs):
        """Log a message to the underlying logger"""

        def _text(arg):
            """Convert a log argument to text"""

            if isinstance(arg, list):
                sep = b',' if arg and isinstance(arg[0], bytes) else ','
                arg = sep.join(arg)

            if isinstance(arg, tuple):
                host, port = arg

                if host:
                    return '%s, port %d' % (host, port) if port else host
                else:
                    return 'port %d' % port if port else 'dynamic port'
            elif isinstance(arg, bytes):
                arg = arg.decode('ascii', errors='replace')

            return arg

        args = [_text(arg) for arg in args]

        return super().log(level, msg, *args, **kwargs)
```

It seems the logger attempts to interpret any tuple it comes across as a host/port pair, which is a problem when it gets called from here (in `asyncssh/sftp.py`):

```python3
    async def put(self, localpaths, remotepath=None, *, preserve=False,
                  recurse=False, follow_symlinks=False,
                  block_size=SFTP_BLOCK_SIZE, max_requests=_MAX_SFTP_REQUESTS,
                  progress_handler=None, error_handler=None):
        """Upload local files

            <snip>
        """

        self.logger.info('Starting SFTP put of %s to %s',
                         localpaths, remotepath)

        await self._begin_copy(LocalFile, self, localpaths, remotepath,
                               preserve, recurse, follow_symlinks, block_size,
                               max_requests, progress_handler, error_handler)
```

The consequence is that if a tuple is passed which isn't of size 2 (and possibly not a string/int pair, though I haven't tested this), this exception gets thrown:

```
ValueError: not enough values to unpack (expected 2, got <number of items in tuple>)
```

**AsyncSSH Version:** 2.2.0

**Python Version:** 3.8.2

**Comment by ronf** [comment]
Thanks very much for reporting this issue! I've checked in a fix in the "develop" branch as commit 88fe0d4 if you'd like to give it a try.

**Comment by Lyrositor** [comment]
Looks good, thanks for the quick response!

**Comment by ronf** [comment]
This fix is now available in AsyncSSH 2.2.1.
